src/bio.c

```diff
@@ -82,12 +73,24 @@ static unsigned int bio_job_to_worker[] = {
     [BIO_RDB_SAVE] = 3,
 };
 
-static pthread_t bio_threads[BIO_WORKER_NUM];
-static pthread_mutex_t bio_mutex[BIO_WORKER_NUM];
-static pthread_cond_t bio_newjob_cond[BIO_WORKER_NUM];
-static list *bio_jobs[BIO_WORKER_NUM];
+typedef struct {
+    const char *const bio_worker_title;
+    pthread_t bio_thread_id;
+    pthread_mutex_t bio_mutex;
+    pthread_cond_t bio_newjob_cond;
+    list *bio_jobs;
+} bio_worker_data;
+
+static bio_worker_data bio_workers[] = {
+    {"bio_close_file"},
+    {"bio_aof"},
+    {"bio_lazy_free"},
+    {"bio_rdb_save"}};
+static const bio_worker_data *const bio_worker_end = bio_workers + (sizeof bio_workers / sizeof *bio_workers);
+#define BIO_WORKER_NUM(X) ((X) - bio_workers)
+
 static unsigned long bio_jobs_counter[BIO_NUM_OPS] = {0};
-static __thread unsigned long bio_thread_id = 0;
+static _Thread_local size_t bio_worker_num = 0;
```

Comment:
I like the use of c11 thread local. maybe we should use this PR to also replace all the existing __thread uses (not many) as we are c11 compatible and although we do only support gcc/clang it would be nice to remove this dependency